gvfs-helper.c

```diff
@@ -2820,10 +2820,13 @@ static void do_req(const char *url_base,
 		 * with the main Git server (because the cache-server
 		 * doesn't handle 401s).
 		 *
-		 * TODO Think about if we really need to handle this case.
-		 * TODO Guard with "if (params->sever_type == __MAIN)"
+		 * Set an empty u/p to get CURL to automatically negotiate
```

Comment:
Source context (because GitHub doesn't let you comment on lines outside the default context): this change is in the `else` part of the `if (creds && creds->username)` control flow statement.

What happens if `creds->username` is `NULL` and `creds->password != NULL`?
We'll fall into this `CURLAUTH_ANY` flow.. what if users have manually specified a PAT as the password without a username? I.e., `":password123"`